### PR TITLE
CanIgnoreReturnValueSuggester: Support exempting method annotations

### PR DESCRIPTION
**Pull request by rickie**
While there, exempt `@AfterTemplate` methods from analysis.

### CODE REVIEW

**Comment by rickie** [comment]
Suggested commit message:
```
CanIgnoreReturnValueSuggester: Support custom exempting method annotations

While there, exempt `@AfterTemplate` methods from analysis.
```

File: core/src/main/java/com/google/errorprone/bugpatterns/checkreturnvalue/CanIgnoreReturnValueSuggester.java [hunk 1]
```diff
@@ -76,19 +77,39 @@ public final class CanIgnoreReturnValueSuggester extends BugChecker implements M
...
+    // If the method is already directly annotated w/ @CRV, @CIRV, or @AfterTemplate, bail out.
+    if (hasAnnotation(methodSymbol, CRV, state)
+        || hasAnnotation(methodSymbol, CIRV, state)
+        || hasAnnotation(methodSymbol, REFASTER_AFTER_TEMPLATE, state)) {
+      return Description.NO_MATCH;
+    }
+
+    // If the method is annotated with an annotation that is exempted, bail out.
+    if (additionalExemptingMethodAnnotations.stream()
+        .anyMatch(annotation -> ASTHelpers.hasAnnotation(methodTree, annotation, state))) {
```
Comment (Stephan202) [inline core/src/main/java/com/google/errorprone/bugpatterns/checkreturnvalue/CanIgnoreReturnValueSuggester.java]: We can:
1. Statically import this method.
2. Use the `MethodSymbol`-accepting overload used above.
3. Use this `anyMatch` also for the cases mentioned above.

**Review by Stephan202 — APPROVED** [review]
Rebased and added a commit. One can debate whether `@CRV` and `@CIRV` are "exempting" method annotations, but they do prevent this checker for making a suggestion :shrug:

Slightly tweaked the suggested commit message.

**Comment by rickie** [comment]
Upstream PR: https://github.com/google/error-prone/pull/4009.

**Comment by Stephan202** [comment]
Upstream PR was merged :+1:

**Pull request closed without merging**